packages/types/src/envelope.ts

```diff
@@ -101,7 +101,7 @@ export type ProfileItem = BaseEnvelopeItem<ProfileItemHeaders, Profile>;
 export type ProfileChunkItem = BaseEnvelopeItem<ProfileChunkItemHeaders, ProfileChunk>;
 export type SpanItem = BaseEnvelopeItem<SpanItemHeaders, Partial<SpanJSON>>;
 
-export type EventEnvelopeHeaders = { event_id: string; sent_at: string; trace?: DynamicSamplingContext };
+export type EventEnvelopeHeaders = { event_id: string; sent_at: string; trace?: Partial<DynamicSamplingContext> };
```

Comment:
sorry, forgot to mention this here: this was just not correctly annotated before, really. Since we did not type `dynamicSamplingContext` on the `sdkProcessingMetadata` before, this worked by accident. but because this is really `Partial<DSC>` (based on what we set on it, which is also partial), it makes this necessary to be correct.

We could also choose to leave this and cast it in the place where we set this, but really that means lying to ourselves because we don't _know_ it is complete. 🤷 no strong feelings though, if we feel this is a bad type change I can also cast it.